cardinal/component/storage.go

```diff
@@ -0,0 +1,6 @@
+package component
+
+type StorageProvider interface {
+	GetSchema(componentName string) ([]byte, error)
+	SetSchema(componentName string, schemaData []byte) error
+}
```

Comment:
My original vision for this is that it is likely we might want to store more things beyond schema.

That said, after thinking about it, narrow interfaces seems better so I'll go ahead and rename this to SchemaProvider as you want.

Reply:
on a second thought, Im renaming this to SchemaStorage because I think SchemaStorageProvider is mouthful and I still want to make it clear that "hey this thing should be backed by a storage like struct"